src/DataValues/DecimalValue.php

```diff
@@ -360,13 +350,20 @@ public function getArrayValue() {
 	}
 
 	/**
-	 * Constructs a new instance of the DataValue from the provided data.
-	 * This can round-trip with @see getArrayValue
+	 * Constructs a new instance from the provided data. Required for @see DataValueDeserializer.
+	 * This is expected to round-trip with @see getArrayValue.
+	 *
+	 * @deprecated since 0.8.3. Static DataValue::newFromArray constructors like this are
+	 *  underspecified (not in the DataValue interface), and misleadingly named (should be named
+	 *  newFromArrayValue). Instead, use DataValue builder callbacks in @see DataValueDeserializer.
```

Comment:
I understand that. I do not oppose to depreciation or anything. I am only mentioning that I found the depreciation notice a bit confusing (a single sentence there, in particular).

I don't know this all DataValues stuff well, so it would be the best if @JeroenDeDauw or @brightbyte chimed in here I guess. It simply seems a bit odd to me that this components mentions some other component which is not even its dependency etc.